src/go/intelligence/pipeline/delete_old_silents.go

```diff
@@ -0,0 +1,112 @@
+package pipeline
+
+import (
+	"errors"
+	"go/intelligence/models"
+	"go/intelligence/utils"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"time"
+)
+
+// Module     : intelligence
+// Author     : Rahul Indra <indrarahul2013 AT gmail dot com>
+// Created    : Wed, 1 July 2020 11:04:01 GMT
+// Description: CMS MONIT infrastructure Intelligence Module
+
+//DeleteSilence - function for deleting expired silences
+func DeleteSilence(data <-chan models.AmJSON) <-chan models.AmJSON {
+
+	finalData := make(chan models.AmJSON)
+	go func() {
+		deleteSilenceHelper()
+		for each := range data {
+			finalData <- each
+		}
+		close(finalData)
```

Comment:
this channel may not be closed if above code fails and exit prematurely. It is much safer to use `defer close(finalData)` at the beginning of this function. The defer call will ensure that channel is closed regardless of body function execution.

Reply:
Resolved.